apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.js

```diff
@@ -466,616 +453,148 @@ async function writeEdit(fileEdit) {
   return fileID;
 }
 
-function saveAndSync(fileEdit, locals, callback) {
-  const options = {
-    course_id: locals.course.id,
-    user_id: locals.user.user_id,
-    authn_user_id: locals.authz_data.authn_user.user_id,
+async function saveAndSync(fileEdit, locals) {
+  const serverJob = await createServerJob({
+    courseId: locals.course.id,
+    userId: locals.user.user_id,
+    authnUserId: locals.authz_data.authn_user.user_id,
     type: 'sync',
     description: 'Save and sync an in-browser edit to a file',
-    courseDir: locals.course.path,
-  };
+  });
 
-  serverJobs.createJobSequence(options, (err, job_sequence_id) => {
-    // Return immediately if we fail to create a job sequence
-    if (ERR(err, callback)) return;
+  await serverJob.execute(async (job) => {
```

Comment:
Because we rely on file edits completing "synchronously" during the request; we can't send a response til it's done. See https://github.com/PrairieLearn/PrairieLearn/issues/1631 for ideas on changing that.

Reply:
Oh right, thanks! And #1631 is a great reminder - it would be fantastic to have lazy sync in the background!